File: DesignPatternPython/Behavioral/mediator/user_impl.py
from Behavioral.mediator.user import User

class UserImpl(User):

    def __init__(self, name) -> None:
        super().__init__(name)

    def send(self, msg):
        print("-----")
        print("%s is sending message:%s"%(self.name, msg))
        self.mediator.sendMessage(msg, self)

    def receive(self, msg):
        print("%s receive the message:%s" %(self.name, msg))

File: DesignPatternJava/src/behavioral/mediator/UserImpl.java
package behavioral.mediator;

/**
 * @author tamdx
 */
public class UserImpl extends User {
    public UserImpl(String name) {
        super(name);
    }


    @Override
    public void send(String msg) {
        System.out.println("----");
        System.out.println(this.getName() + " is sending message:" + msg);
        mediator.sendMessage(msg, this);
    }

    @Override
    public void receive(String msg) {
        System.out.println(this.getName() + " received the message :"+msg);
    }
}
